module-bluetooth/Bluetooth/interface/profiles/PhoneInterface.cpp

```diff
@@ -9,17 +9,12 @@ namespace bluetooth
 {
     bool CellularInterfaceImpl::answerIncomingCall(sys::Service *service)
     {
-        callActive = true;
         return CellularServiceAPI::AnswerIncomingCall(service);
     }
 
     bool CellularInterfaceImpl::hangupCall(sys::Service *service)
     {
-        if (callActive) {
```

Comment:
seems like a workaround to me, do we know why was it applied in the first place?

Reply:
That flag was added to prevent showing "end call" screen when some device sent hangup call message during connecting. As far as I understand, this shouldn't be an issue after my changes. If the BTStack's call state machine is handled properly (which I believe now is :P ) sending hangup call message when the call is not established will just be ignored.

Reply:
Thanks for answering!